### PR TITLE
libavcodec/qsvenc_hevc: add main10sp support to hevc_qsv

### PR DESCRIPTION
**Pull request by wenbinc-Bin**
Main10sp is a combination of Main10 and one_pic_only flag.
This profile encode 10bit single still picture.
Add main10sp support to ffmpeg-qsv.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

### CODE REVIEW

File: libavcodec/qsvenc_hevc.c [hunk 1]
```diff
@@ -242,6 +242,7 @@ static const AVOption options[] = {
     { "mainsp",  NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_MAINSP  }, INT_MIN, INT_MAX,     VE, "profile" },
     { "rext",    NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_REXT    }, INT_MIN, INT_MAX,     VE, "profile" },
 
+    { "main10sp", "This profile allow to encode 10 bit single still picture", OFFSET(qsv.main10sp), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VE},
```
Comment (xuguangxin) [inline libavcodec/qsvenc_hevc.c:245]: why it's a bool? not  a profile like mainsp?

**Review by xuguangxin — COMMENTED** [review]
Do we need check input formation for main10 sp?

File: libavcodec/qsvenc_hevc.c (hunk 1 above)
Comment (wenbinc-Bin) [reply libavcodec/qsvenc_hevc.c:245]: Main10sp is not an enum of profile (there is no enumeration like MFX_PROFILE_HEVC_MAIN10SP), so it cannot be added to "profile" option like mainsp.

**Comment by wenbinc-Bin** [comment]
> Do we need check input formation for main10 sp?

I think no, because main10sp use main10 profile. It will check input just like what main10 does.

File: libavcodec/qsvenc_hevc.c (hunk 1 above)
Comment (xuguangxin) [reply libavcodec/qsvenc_hevc.c:245]: do we have the main still picture or the main 12/16 still picture?

Comment (wenbinc-Bin) [reply libavcodec/qsvenc_hevc.c:245]: we have mainsp but don't have main 12/16 sp.
HM also process main10sp in this way - using main10 profile and enable one_pic_only flag.
According to specs, main10sp should set general_profile_idc to 2, which equals to main10. I think this is why they don't set a profile enumeration for main10sp.

**Comment by xhaihao** [comment]
> The Main10sp support in VPL source code:
"https://github.com/oneapi-src/oneVPL-intel-gpu/blob/
d637669768bd8baeeb262ca9cd660687dee2df83/_studio/mfx_lib/
encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp#L4703"

Did you this is an oneVPL feature ?

**Comment by wenbinc-Bin** [comment]
> > The Main10sp support in VPL source code:
> > "https://github.com/oneapi-src/oneVPL-intel-gpu/blob/
> > d637669768bd8baeeb262ca9cd660687dee2df83/_studio/mfx_lib/
> > encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp#L4703"
> 
> Did you this is an oneVPL feature ?

This option is alreaded added as an option to mfx_transcoder.

**Comment by wenbinc-Bin** [comment]
Any other feedback?

**Comment by xhaihao** [comment]
> This option is alreaded added as an option to mfx_transcoder.

If this is not a vpl specific feature, it would be better to update the commit log, otherwise user is easily misled by this log.

**Comment by wenbinc-Bin** [comment]
> If this is not a vpl specific feature, it would be better to update the commit log, otherwise user is easily misled by this log.

Ok, I have updated the commit log.

**Comment by xuguangxin** [comment]
@wenbinc-Bin  are we ready to merge this?
thanks

**Comment by wenbinc-Bin** [comment]
Any other comments on this PR? If no, I will submit to upstream.

**Comment by wenbinc-Bin** [comment]
Hi @xhaihao. Any other comment on this PR. This is a VPL specific feature. Should I submit it to upstream? because oneVPL patches haven't been merged into upstream.

**Comment by xhaihao** [comment]
@wenbinc-Bin Please submit a PR for cartwheel ffmpeg for this oneVPL feature.

**Comment by wenbinc-Bin** [comment]
> @wenbinc-Bin Please submit a PR for cartwheel ffmpeg for this oneVPL feature.

Ok

**Comment by wenbinc-Bin** [comment]
I found that if I use libmfx to load VPL's runtime (libmfx-gen.so), I can enable this feature. I will add explanation to commit message and submit to upstream.

**Pull request closed without merging**